src/test/java/nl/ordina/jobcrawler/scrapers/YachtVacancyScraperTest.java

```diff
@@ -0,0 +1,225 @@
+package nl.ordina.jobcrawler.scrapers;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.exc.MismatchedInputException;
+import nl.ordina.jobcrawler.model.Skill;
+import nl.ordina.jobcrawler.model.Vacancy;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.select.Elements;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.client.RestTemplate;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.List;
+import java.util.Set;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.Mockito.*;
+
+@RunWith(MockitoJUnitRunner.class)
+public class YachtVacancyScraperTest {
+
+    @InjectMocks
+    private YachtVacancyScraper yachtVacancyScraper;
+
+    @Mock
+    private RestTemplate restTemplateMock;
+
+    @Mock
+    private Document documentMock;
+
+    private static Document vacancyDoc;
+    private static Document removedVacancyDoc;
+
+    private static ResponseEntity<YachtVacancyResponse> jsonResponse;
+    private static ResponseEntity<YachtVacancyResponse> noDataResponse;
+
+    @BeforeClass
+    public static void init() throws Exception {
+        // Grab the file content of the files mentioned below. Parse this file using jsoup as HTML.
+
+        File vacancyDocHtml = getFile("/yacht/yachtvacancy.html");
+        vacancyDoc = Jsoup.parse(vacancyDocHtml, "UTF-8");
+
+        File removedVacancyHtml = getFile("/yacht/yachtremovedvacancy.html");
+        removedVacancyDoc = Jsoup.parse(removedVacancyHtml, "UTF-8");
+
+        // Saved .json response in resources folder is being used in this test. Content of this file is needed.
+        File jsonFile = getFile("/yacht/getRequestResponse.json");
+        // We need to map the data from the jsonFile according to our YachtVacancyResponse.class
+        YachtVacancyResponse yachtVacancyResponse = new ObjectMapper().readValue(jsonFile, YachtVacancyResponse.class);
+        jsonResponse = new ResponseEntity<>(yachtVacancyResponse, HttpStatus.OK);
+
+        File jsonFileNoData = getFile("/yacht/getRequestResponseNoData.json");
+        YachtVacancyResponse yachtVacancyResponseNoData = new ObjectMapper().readValue(jsonFileNoData, YachtVacancyResponse.class);
+        noDataResponse = new ResponseEntity<>(yachtVacancyResponseNoData, HttpStatus.OK);
+    }
+
+    @Test
+    public void scrapeVacancies_should_return_YachtVacancyResponse() throws IOException {
```

Comment:
I think the naming convention of test method names is "testMethodName". 
To add more description, you can add a sentence to the assert statement:
```
assertEquals(1, 1+0, "should be equal to one");
```
You can also add a description of the test with the annotation using```@DisplayName("message")```.
[This video](https://app.pluralsight.com/course-player?clipId=9f06b82f-17db-4e0d-bb33-30ea3fd40744) explains it. [This repo](https://github.com/geekcap-pluralsight/productservice.git) also contains an example spring application including testing.

Reply:
@brampat what is the naming convention of test method names?

Reply:
The 'should' added to the method name is a convention that's being used in Angular. No idea what or if there is any convention for Java, but to me it is very clear what the test 'should' do/return.

Reply:
I'ld say there's no single way of doing this (correctly), so it's more of a pick one we agree on and stick with that. I would like to pick one that is compact in JUnit since you're doing JUnit and not Groovy / Spock. This means you are limited to what Java would allow, while Spock van do stuff like this:

`
@Unroll
def "maximum of #a and #b is #c"() { 

...

        where:
        a | b || c
        3 | 5 || 5
        7 | 0 || 7
        0 | 0 || 0
 }
`

Which would result in output like this:
`
maximum of 3 and 5 is 5   PASSED
maximum of 7 and 0 is 7   FAILED

Math.max(a, b) == c
    |    |  |  |  |
    |    7  0  |  7
    42         false

maximum of 0 and 0 is 0   PASSED
`

Here are some resources which outline different options:
* https://medium.com/@stefanovskyi/unit-test-naming-conventions-dd9208eadbea
* https://dzone.com/articles/7-popular-unit-test-naming